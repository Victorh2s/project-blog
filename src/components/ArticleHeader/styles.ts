import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`

  text-align: center;
  justify-content: center;
  margin-bottom: ${theme.spacings.xlarge};
  border-bottom: 0.1rem solid ${theme.colors.darkeyGray};

  ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }

  @media ${theme.media.lteMedium} {
      ${HeadingStyles}{
      font-size: ${theme.font.sizes.large};
    }
  }

`}
`;

export const Text = styled.p`
  text-align: center;
  justify-content: center;
`;

export const Img = styled.img`
  max-width: 100%;
  display: block;
`;
