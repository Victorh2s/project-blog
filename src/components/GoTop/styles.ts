import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    color: ${theme.colors.primary};
    border: 0.1rem solid;
    border-radius: 50rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
