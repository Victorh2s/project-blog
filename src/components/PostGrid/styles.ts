import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  width: 100%;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
  max-width: 120rem;
  max-height: 300rem;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
  gap: ${theme.spacings.large};
  padding: ${theme.spacings.large};
  text-overflow: ellipsis;

  @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
${({ theme }) => css`

    padding: ${theme.spacings.large};
    text-align: center;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.secondary};

`}
`;
