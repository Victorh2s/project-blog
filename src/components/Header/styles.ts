import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as LogoLinkstyles } from '../LogoLink/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: 96rem;
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    ${LogoLinkstyles}{
      >img{
        border: 0.3rem solid;
        border-color:  rgb(74 222 128);
      }
    }

    @media ${theme.media.lteMedium}{
      display:flex;
      flex-flow: column wrap;

      & > ${HeadingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }

    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.lteMedium}{
      margin-left: 0;
    }
  `}
`;
