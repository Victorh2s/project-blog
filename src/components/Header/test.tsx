import props from './mock';
import { screen } from '@testing-library/react';
import { describe, it } from 'vitest';
import { Header } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<Header/>', () => {
  it('Should render Header with a logo, heading and textcomponent', () => {
    renderTheme(<Header {...props} showText={true} />);
    expect(
      screen.getByRole('heading', { name: props.BlogName }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /Usuário teste/i })).toHaveAttribute(
      'src',
      props.logo,
    );
    expect(screen.getByText(props.BlogDescription)).toBeInTheDocument();
  });

  it('Should render only Image', () => {
    renderTheme(<Header {...props} showText={false} />);
    expect(screen.queryByRole(props.BlogName)).not.toBeInTheDocument();
    expect(screen.getByRole('img', { name: /Usuário teste/i })).toHaveAttribute(
      'src',
      props.logo,
    );
    expect(screen.queryByRole(props.BlogDescription)).not.toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(
      <Header {...props} showText={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
