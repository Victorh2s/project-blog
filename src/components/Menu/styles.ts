import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  Visible: boolean;
};

const ContainerChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? 0 : '-35rem'};
  overflow-y: ${Visible ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '0' : '-35rem'};
  }
`;

export const Container = styled.div<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${ContainerChanger(Visible, theme)}
  `}
`;

export const Nav = styled.nav`
    margin: 0 auto;
    width: 100%;
`;

export const MenuLinks = styled.div`

   margin-bottom: 5rem;

`;

export const Logo = styled.div`

    ${HeadingStyles} {

      img{
        margin: 0 auto;
        margin-bottom: 3rem;
      }
    }

`;

const ButtonChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? '26rem,' : '1rem'};
  color: ${Visible ? theme.colors.secondary : theme.colors.primary};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '26rem' : '-0.5rem'};
  }
`;

export const Button = styled.a<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    left: 26rem;
    color: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    transition: all 300ms ease-in-out;
    ${ButtonChanger(Visible, theme)}
  `}
`;
