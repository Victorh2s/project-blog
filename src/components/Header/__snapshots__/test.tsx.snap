// Vitest Snapshot v1

exports[`<Header/> > Should match snapshot 1`] = `
.c2 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c6 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c4 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
  background: #333333;
}

.c7 {
  font-size: 1.6rem;
}

.c7 p {
  margin: 4.0rem0;
}

.c7 a,
.c7 a:visited,
.c7 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c7 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c7 pre {
  background: #0A1128;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 4.0rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c7 img {
  max-width: 100%;
}

.c7 .image {
  background: #DDDDDD;
  line-height: 0;
}

.c7 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c7 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem;
}

.c7 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c7 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c7 table {
  width: 100%;
  border-collapse: collapse;
}

.c7 table td,
.c7 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c7 ul,
.c7 ol {
  margin: 4.0rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 5.6rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c0 .c1 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c0 .c3 >img {
  border: 0.3rem solid;
  border-color: rgb(74 222 128);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

@media (max-width:768px) {
  .c7 {
    font-size: 1.6rem;
  }

  .c7 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
  }

  .c0 > .c1 {
    margin: 0 0 2.4rem 0;
  }
}

@media (max-width:768px) {
  .c5 {
    margin-left: 0;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      <a
        class="c3 c4"
        href="/"
        target="_blank"
      >
        <img
          alt="Usuário teste - Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut."
          src="assets/images/logo.png"
        />
      </a>
    </h1>
    <div
      class="c5"
    >
      <h1
        class="c1 c6"
      >
        Usuário teste
      </h1>
      <div
        class="c7"
      >
        Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut.
      </div>
    </div>
  </header>
</div>
`;
